source/src/molecule-unfolding/cdk/utils/custom-resource.ts

```diff
@@ -0,0 +1,201 @@
+import * as cdk from 'aws-cdk-lib';
+import {
+  Stack
+} from 'aws-cdk-lib';
+import {
+  Construct
+} from 'constructs'
+
+import {
+  aws_iam as iam
+} from 'aws-cdk-lib'
+
+import {
+  aws_ec2 as ec2
+} from 'aws-cdk-lib'
+
+import {
+  NodejsFunction
+} from 'aws-cdk-lib/aws-lambda-nodejs';
+
+import {
+  Runtime
+} from 'aws-cdk-lib/aws-lambda';
+
+import {
+  Provider
+} from 'aws-cdk-lib/custom-resources';
+
+
+function createCustomResourceLambdaRole(scope: Construct, roleName: string): iam.Role {
+  const account_id = Stack.of(scope).account
+
+  const role = new iam.Role(scope, `${roleName}`, {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  role.addToPolicy(new iam.PolicyStatement({
+    effect: iam.Effect.ALLOW,
+    resources: [
+      '*'
+    ],
+    actions: [
+      "ec2:CreateNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface",
+      "ec2:AssignPrivateIpAddresses",
+      "ec2:UnassignPrivateIpAddresses"
+    ]
+  }));
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: [
+      `arn:aws:logs:*:${account_id}:log-group:*`
+    ],
+    actions: [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:CreateLogGroup"
+    ]
+  }));
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: [
+      `arn:aws:iam::${account_id}:role/*`,
+      `arn:aws:iam::${account_id}:policy/*`
+    ],
+    actions: [
+      "iam:DetachRolePolicy",
+      "iam:DeleteRolePolicy",
+      "iam:CreateRole",
+      "iam:DeleteRole",
+      "iam:AttachRolePolicy",
+      "iam:UpdateRole",
+      "iam:PutRolePolicy",
+      "iam:CreatePolicy",
+      "iam:DeletePolicy",
+      "iam:GetRole",
+      "iam:GetPolicy",
+      "iam:PassRole",
+    ]
+  }));
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: [
+      `arn:aws:events:*:${account_id}:rule/*`
```

Comment:
could we limit to region `us-west-2` and certain event rule?

Reply:
will add us-west-2